File: app/src/main/java/com/scatl/uestcbbs/module/post/adapter/CreatePostPollAdapter.java
package com.scatl.uestcbbs.module.post.adapter;

import android.view.View;
import android.widget.ImageView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;

/**
 * author: sca_tl
 * description:
 * date: 2020/2/16 15:58
 */
public class CreatePostPollAdapter extends BaseQuickAdapter<String, BaseViewHolder> {
    public CreatePostPollAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, String item) {
        helper.setText(R.id.item_create_post_poll_textview, (helper.getLayoutPosition() + 1) + ". " + item);

    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/post/adapter/CreatePostPollAdapter.kt
package com.scatl.uestcbbs.module.post.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemCreatePostPollBinding
import com.scatl.uestcbbs.helper.PreloadAdapter

/**
 * created by sca_tl at 2023/6/10 11:19
 */
class CreatePostPollAdapter: PreloadAdapter<String, ItemCreatePostPollBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemCreatePostPollBinding {
        return ItemCreatePostPollBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int, item: String?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }
        val binding = holder.binding as ItemCreatePostPollBinding
        binding.itemCreatePostPollTextview.text = (holder.layoutPosition + 1).toString() + ". " + item
    }
}